Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FlatPlatePaalmanPingsCorrection.py

```diff
@@ -619,35 +619,34 @@ def _can_thickness_calc(self, can_x_section, can_x_section_efixed, can_thickness
 
     def _container_transmission_calc(self, acc, acc1, acc2, ki_s, kf_s, ki_c1, kf_c2, ass):
         if self._has_can_front_in and self._has_can_back_in:
-            acc = self._can_front_thickness * acc1 * np.exp(-kf_c2)
-            acc += self._can_back_thickness * acc2 * np.exp(-ki_c1)
-            acc /= (self._can_front_thickness + self._can_back_thickness)
+            acc = (self._can_front_thickness * acc1 * np.exp(-kf_c2) + self._can_back_thickness * acc2 * np.exp(-ki_c1)) \
+                  / (self._can_front_thickness + self._can_back_thickness)
+            if self._has_sample_in:
+                acsc = (self._can_front_thickness * acc1 * np.exp(-kf_s - kf_c2) +
+                        self._can_back_thickness * acc2 * np.exp(-ki_c1 - ki_s)) / \
+                       (self._can_front_thickness + self._can_back_thickness)
```

Comment:
This is a nice re-arrangement :smile:, it is clear how each term is used in calculating each factor.
There is a further way to prevent recalculating the same values on multiple occasions:

The acc factor appears to be the sum of two product terms. These same product terms are recalculated for the acsc.
i.e.
let `x = self._can_front_thickness * acc1 * np.exp(-kf_c2)` (the first term used to calculate the acc correction factor)
then for the calculation of the ascs factor: 
`self._can_front_thickness * acc1 * np.exp(-kf_s - kf_c2)` can be rewritten as `x * np.exp(-kf_s)`

Similarly for the second term,
let `y = self._can_back_thickness * acc2 * np.exp(-ki_c1)` (the second term used to calculate the acc correction factor)
then in calculation of the ascs factor: 
`self._can_back_thickness * acc2 * np.exp(-ki_s - ki_c1)` can be rewritten as `y * np.exp(-ki_s)`

This could also allow for using pre-calculated `np.exp(-ki_s)` and `np.exp(-kf_s)` terms, which are used later in this function.

Reply:
You are right in saying this could be simplified this way. However, I am not sure the gains are worth while. In either case you have to calculate the exponential of a numpy array, which will be the slow part. Doing the array subtraction such as `-kf_s - kf_c2` should be quick in comparison.

I think I would rather leave it as it is, so that it is more readable, as the code is already a bit hard to follow. As far as I understand the arrays would normally be small, thought perhaps there are exceptions to this.